sources/SilkTouch/SilkTouch/Program.cs

```diff
@@ -62,46 +72,98 @@
     // Register MSBuild
     MSBuildLocator.RegisterDefaults();
 
+    JobContext jobContext = new();
+
     var sp = new ServiceCollection()
+        .AddSingleton(jobContext)
+        .AddSingleton<IProgressService, ProgressService>()
```

Comment:
Perhaps this line should be part of AddSilkTouch. Note that you'll probably have to move the ConsoleRenderer AddSingleton call to after AddSilkTouch (might not though, but if it breaks this might be why).